[HusseinSafwan02] iP

HusseinSafwan02 opened the pull request:
### ChillChief

> "Simplicity boils down to two steps: identify the essential, eliminate the rest."– Leo Babauta [(source)](https://dansilvestre.com/productivity-quotes/)

ChillChief is your go to assistant 'bro' for managing tasks in your life. It is

- Super **_Chill_**
- Super _Easy_
- Super _Useful_

All you need to do is,

1. Download it from [here](https://github.com/HusseinSafwan02/ip/releases/tag/A-Jar).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**

Features:

- [ ] _Add_, _Delete_, and _Find_ your tasks
- [ ] Manage _Deadlines_, _Events_ and _Todo_'s
- [ ] Receive _Reminders_ (coming soon!)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

HusseinSafwan02 committed:
Rename, Greet and Exit

HusseinSafwan02 committed:
Echo and "bye"

HusseinSafwan02 committed:
Added ability to store tasks

HusseinSafwan02 committed:
Tasks can be marked done

HusseinSafwan02 committed:
Specific task types, dates

HusseinSafwan02 committed:
Automated Text-UI-Testing

HusseinSafwan02 committed:
Handle Errors

HusseinSafwan02 committed:
Support for deleting tasks

HusseinSafwan02 committed:
Saving feature

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
clean up

HusseinSafwan02 committed:
Merge branch 'branch-level-7'

Teeesa7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
 public class Duke {
+
+    // confirmation comment after adding a task
+    public static String addComment (Task task, int n) {
```
Shouln't there be no space following ```addComment``` and the arguments it takes in?

Teeesa7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+        return message;
+    }
+
+    public static String introMessage() {
```
Perhaps another name for the method? Something like ```getIntroMessage()```?, as ```introMessage()``` sounds like a String variable instead 😅

Teeesa7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+        return greeting + mickey + intro;
+    }
+
+    public static String outroMessage() {
```
Please refer to the comment given for ```introMessage()```

Li-Zizhen replied:
Perhaps it's better to make the method name more understandable.

Teeesa7 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task{
```
There should be a space after ```Task``` and ```{``` right? I noticed the same issue in other places, so just mind that :)

Teeesa7 commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,84 @@
+
+    private Task parseTask(String data) {
+        String[] parts = data.split(" \\| ");
+        if (parts.length < 3) return null; // Basic validation
```
Maybe you can use the K&R style brackets (Egyptian style) brackets here for better readability?

Teeesa7 reviewed (commented):
LGTM, just a tiny issue of unnecessary spaces at different places in the code.

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+                        System.out.println(addComment(tasks.get(count - 1), count));
+                        fileManager.saveTasks(tasks);
+                    }
+
```
Not sure if this empty line is left on purpose or not.

Li-Zizhen commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+        return this.time;
+    }
+
+    @Override public String toString() {
```
Better to put Override above the method.

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+    public static void listTasks(ArrayList<Task> taskList, int count) {
+        System.out.print("____________________________________________________________\r\n");
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 0; i < taskList.size(); i++ ) {
```
Better to delete space after ++

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+        return message;
+    }
+
+    public static String replacer(String input) {
```
Better to use a verb for method naming.

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+                boolean isUnmark = userInputArray[0].equals("unmark");
+                char lastChar = userInput.charAt(userInput.length() - 1);
+
+
```
better not to leave so many empty lines.

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+                    if (isEvent) {
+                        String task = getTask(userInput);
+                        int index = userInput.indexOf("/");
+                        String when = replacer(userInput).substring(index);
```
Better use a noun to name variable.

Li-Zizhen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,165 @@
+                    }
+                    if (isDeadline) {
+                        int index = userInput.indexOf("/");
+                        String when = userInput.substring(index + 3);
```
same as above

Li-Zizhen commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    private Task parseTask(String data) {
+        String[] parts = data.split(" \\| ");
```
Perhaps rename parts to something more understandable?

Li-Zizhen reviewed (commented):
Overall, your code is easy to understand. Most of your code follows the coding standard. However, there are some minor things you may fix. In all, LGTM, good job!

HusseinSafwan02 committed:
Dates and Times

HusseinSafwan02 committed:
Clean Up

HusseinSafwan02 committed:
Implemented TextUi class to manage messages

HusseinSafwan02 committed:
TaskList class to store/delete tasks

HusseinSafwan02 committed:
Modification of tasks to handle own fileStrings

HusseinSafwan02 committed:
Implemented toFileString() method for each task

HusseinSafwan02 committed:
Separated TextUi class to handle Ui

HusseinSafwan02 committed:
TaskList class for encapsulation of tasklist

HusseinSafwan02 committed:
Finalised OOP structure

HusseinSafwan02 committed:
Organised into packages

HusseinSafwan02 committed:
Implemented 'help' command

HusseinSafwan02 committed:
Merge branch 'add-gradle-support'

HusseinSafwan02 committed:
Set up Gradle

HusseinSafwan02 committed:
Unit testing for some non-trivial methods

HusseinSafwan02 committed:
updated build for JavaFX dependencies

HusseinSafwan02 committed:
Some Javdoc comments

HusseinSafwan02 committed:
Completed Javadoc comments.

HusseinSafwan02 committed:
Followed coding standards

HusseinSafwan02 committed:
Implemented "find" functionality.

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
Merge branch 'branch-Level-9'

HusseinSafwan02 committed:
Merge branch 'branch-A-CodingStandard'

HusseinSafwan02 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/chillchief/ChillChief.java

HusseinSafwan02 committed:
Merge branch 'branch-A-JavaDoc'

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
Coding standards

HusseinSafwan02 committed:
Javadoc comments for all public methods

HusseinSafwan02 committed:
Implemented find feature

HusseinSafwan02 committed:
'help' command: to allow users to learn a list of useful commands.

A 'help' command is useful for new users who are using ChillChief
for the first time.

Let's,
* update the parser to handle the 'help' command
* update the TextUi to generate the text Ui for the user

HusseinSafwan02 committed:
Assertions for important assumptions.

No checking of important assumptions at various points in code.

Checking of assumptions through assertions ensures that different
points of the code is functioning as expected.

Assertions added at key points with assumptions in private methods.

Ensures arguments for private methods are valid.

HusseinSafwan02 committed:
Improved Code Quality

HusseinSafwan02 committed:
Merge pull request #2 from HusseinSafwan02/branch-A-CodeQuality

Improved Code Quality

HusseinSafwan02 committed:
Merge pull request #3 from HusseinSafwan02/branch-A-Assertions

Assertions for important assumptions.

HusseinSafwan02 committed:
Implemented 'help' command and more friendly syntax

HusseinSafwan02 committed:
Update README.md

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
No exception thrown from main method

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
no message

HusseinSafwan02 committed:
Screenshot showing full GUI window

HusseinSafwan02 committed:
Finalized README
